server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -48,27 +67,71 @@ public String typeName() {
 
         @Override
         public Query existsQuery(SearchExecutionContext context) {
-            // This query is not performance sensitive, it only helps assess
-            // quality of the data, so we may use a slow query. It shouldn't
-            // be too slow in practice since the number of unique terms in this
-            // field is bounded by the number of fields in the mappings.
             return new TermRangeQuery(name(), null, null, true, true);
```

Comment:
I don't know why the comment was removed, I would leave it.

Reply:
Since we have doc_values for the `_ignored` field, I was hoping this would also have a positive effect on the exists query. We're relying on an exists query on the  `_ignored` field in the new dataset quality page to show users the percentage of documents in their data streams that are degraded i.e. that have ignored fields.

Reply:
right but this comment is on the legacy impl of the exists query, which stays the same, so the comment still applies. The new impl does a plain FieldExistsQuery based on doc_values, and that is definitely better, hence no comment on that one.